wolfcrypt/src/asn.c

```diff
@@ -3460,6 +3460,69 @@ word32 SetBitString(word32 len, byte unusedBits, byte* output)
 #endif /* !NO_RSA || HAVE_ECC || HAVE_ED25519 || HAVE_ED448 */
 
 #ifdef ASN_BER_TO_DER
+
+#define BER_OCTET_LENGTH 4096
+
+/* sets the terminating 0x00 0x00 at the end of an indefinite length
+ * returns the number of bytes written */
+word32 SetIndefEnd(byte* in)
```

Comment:
Can you add a `word32 in_spc` arg or somesuch, and check for overrun, to make this safer?  The ASN subsys has a ton of incumbent code that copies bytes blindly into an output buffer of unknown length, which is a notoriously dangerous code pattern.  Gotta start fixing it somewhere, and it may as well be here.

Reply:
With matching word32 returns to the existing Set* functions, it's not really setup to handle error'ing out. Unless it is more of a silent error out of returning 0 or if this particular function deviates from the existing pattern and returns an int. The size check for most of the ASN Set* functions is done at the beginning of the calling function comparing a "totalSz" value expected to the output buffer size before it reaches a point of calling the Set* functions.